src/test/components/MenuButtons.test.js

```diff
@@ -14,56 +14,210 @@ import {
-      container,
+      ...renderResult,
+      getShareButton,
+      getInnerShareButton,
+      getShareWithUrlsButton,
+      getInnerShareWithUrlsButton,
+      getMenuButtonsContainer,
+      getPermalinkButton,
     };
   }
 
-  // profile.meta.networkURLsRemoved flag is set to false as a default.
-  it('renders the MenuButtons buttons', () => {
-    const { store, container } = setup();
-    store.dispatch(startSymbolicating());
+  describe('share and save buttons', function() {
+    // Mock hash
+    const hash = 'c5e53f9ab6aecef926d4be68c84f2de550e2ac2f';
 
-    // MenuButtons is rendering a fragment with several children. We need to
-    // check all children to assess that the component renders properly.
-    expect(Array.from(container.children)).toMatchSnapshot();
+    beforeAll(function() {
+      (window: any).TextEncoder = TextEncoder;
+    });
 
-    store.dispatch(doneSymbolicating());
-    expect(Array.from(container.children)).toMatchSnapshot();
-  });
+    afterAll(async function() {
+      delete (window: any).TextEncoder;
+    });
 
-  it('renders the MenuButtons buttons with profile.meta.networkURLsRemoved set to true', () => {
-    const profile = processProfile(createGeckoProfileWithJsTimings());
-    profile.meta.networkURLsRemoved = true;
-    const { store, container } = setup(profile);
-    store.dispatch(startSymbolicating());
+    beforeEach(function() {
+      // Flow doesn't know uploadBinaryProfileData is a jest mock.
+      (uploadBinaryProfileData: any).mockImplementation(() =>
+        Promise.resolve(hash)
+      );
+      // Flow doesn't know sha1 is a jest mock.
+      (sha1: any).mockImplementation((_data: Uint8Array) =>
+        Promise.resolve(hash)
+      );
+      // Flow doesn't know shortenUrl is a jest mock.
+      (shortenUrl: any).mockImplementation(() =>
+        Promise.resolve('https://perf-html.io/')
+      );
+    });
 
-    expect(Array.from(container.children)).toMatchSnapshot();
+    // profile.meta.networkURLsRemoved flag is set to false as a default.
+    it('render correctly', () => {
+      const { store, container } = setup();
+      store.dispatch(startSymbolicating());
 
-    store.dispatch(doneSymbolicating());
-    expect(Array.from(container.children)).toMatchSnapshot();
-  });
+      // MenuButtons is rendering a fragment with several children. We need to
+      // check all children to assess that the component renders properly.
+      expect(Array.from(container.children)).toMatchSnapshot();
+
+      store.dispatch(doneSymbolicating());
+      expect(Array.from(container.children)).toMatchSnapshot();
+    });
+
+    it('render buttons with profile.meta.networkURLsRemoved set to true', () => {
+      const profile = processProfile(createGeckoProfileWithJsTimings());
+      profile.meta.networkURLsRemoved = true;
+      const { store, container } = setup(profile);
+      store.dispatch(startSymbolicating());
+
+      expect(Array.from(container.children)).toMatchSnapshot();
```

Comment:
Please remove snapshots here, and create a more specific test. When this type of snapshot test fails it's really hard to know why it failed, and whether it was just markup changing, or actually breaking some assumption.

Reply:
Actually these tests were here before and it's not exactly testing anything because after the test library change, apparently changing the `profile.meta.networkURLsRemoved` value doesn't change anything on snapshot. There is nothing to test without a snapshot since it's all internal. It makes sense to write some unit tests for `profile.meta.networkURLsRemoved` I will do that in a following PR.